src/common/yamlwrapper.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+*		This file is a part of rAthena++.
+*		Copyright(C) 2017 rAthena Development Team
+*		https://rathena.org - https://github.com/rathena
+*
+*	This program is free software: you can redistribute it and/or modify
+*	it under the terms of the GNU General Public License as published by
+*	the Free Software Foundation, either version 3 of the License, or
+*	(at your option) any later version.
+*
+*	This program is distributed in the hope that it will be useful,
+*	but WITHOUT ANY WARRANTY; without even the implied warranty of
+*	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*	GNU General Public License for more details.
+*
+*	You should have received a copy of the GNU General Public License
+*	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*
+*	Author(s)
+*		Jittapan Pluemsumran <secret@rathena.org>
+*/
+
+#include "yamlwrapper.h"
+#include <cstring>
+#include "malloc.h"
+
+extern "C" {
+
+yamlwrapper::yamlwrapper(std::string file) {
+	this->root = YAML::LoadFile(file);
+}
+
+yamlwrapper::yamlwrapper(YAML::Node node) {
+	this->root = node;
+}
+
+yamliterator::yamliterator(YAML::Node sequence) {
+	this->sequence = sequence;
+	this->index = 0;
+}
+
+yamliterator* yamlwrapper::iterator() {
+	return new yamliterator(this->root);
+}
+
+yamlwrapper* yaml_load_file(const char* file_name) {
+	return new yamlwrapper(file_name);
+}
+
+extern "C++" YAML::Node yaml_get_node(YAML::Node& node, std::string& key) {
+	size_t pos = key.find('.');
```

Comment:
Missing nullpointer check for both node and key.
If key is null or empty no 'find' call is necessary.